connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -423,24 +471,29 @@ export async function syncDeltaForRootNode({
 
   const dataSourceConfig = dataSourceConfigFromConnector(connector);
 
-  const { nodeType } = typeAndPathFromInternalId(rootNodeId);
+  const nodeTypes = parentIds.map(
+    (nodeId) => typeAndPathFromInternalId(nodeId).nodeType
+  );
 
-  if (nodeType !== "drive" && nodeType !== "folder") {
-    throw new Error(`Node ${rootNodeId} is not a drive or folder`);
+  if (
+    nodeTypes.some((nodeType) => nodeType !== "drive" && nodeType !== "folder")
+  ) {
+    throw new Error(`Some of ${parentIds} are not a drive or folder`);
   }
 
-  const node = await MicrosoftNodeResource.fetchByInternalId(
-    connectorId,
-    rootNodeId
+  const nodes = await Promise.all(
```

Comment:
use `fetchByInternalId**s**` here to have only one db call
(compare length below to check for missing)
(or refactor fetch to have a parameter to throw if missing)